pkg/download/download.go

```diff
@@ -220,13 +221,22 @@ func createConfigsFromAPI(
 		if filter {
 			continue
 		}
+		emptyfolder = false
 		ycreator.AddConfig(cleanName, name)
 	}
 
-	err = ycreator.CreateYamlFile(fs, subPath, api.GetId())
-	if err != nil {
-		util.Log.Error("error creating config api yaml file: %v", err)
-		return err
+	if emptyfolder {
```

Comment:
Could you add a test for the logic please? 

There is some limited test setup in `download_test.go` that should give some pointers on how you might setup that test. 

In general see 
* https://dynatrace-oss.github.io/dynatrace-monitoring-as-code/Get-started/intro
* https://gobyexample.com/testing
on Go testing :)